src/NHibernate/Dialect/Dialect.cs

```diff
@@ -1829,6 +1926,66 @@ public virtual string[] UnQuote(string[] quoted)
 			return unquoted;
 		}
 
+		/// <summary>
+		/// Convert back-tilt quotes in a name for being used as an aliasname.
+		/// </summary>
+		/// <param name="aliasName">Name of the alias.</param>
+		/// <returns>A name with back-tilt quotes converted if any.</returns>
+		public virtual string ConvertQuotesForAliasName(string aliasName)
```

Comment:
Having X methods doing the same isn't necessary.

Reply:
This is done for consistency with the `QuoteFor` series, which does exactly the same. I think the idea is to allow a dialect to override some but not all of them with a different implementation, in case a database may have different quoting rules depending of the quoted object type.

If we consider this as not necessary, then we should obsolete the `QuoteFor` series by the way and keep only a general purpose `QuoteFor`.

We may improve the pattern by calling the corresponding `QuoteFor` from `ConvertQuotesFor`, in order to allow to override only one method instead of both. (But this will cause the convert to do the additional check done in quote for, which it does not need.)